src/Clock.tsx

```diff
@@ -0,0 +1,74 @@
+import { Component } from 'react';
+
+function getRandomName(): string {
+  const value = Date.now().toString().slice(-4);
+
+  return `Clock-${value}`;
+}
+
+type Props = {
+  name: string;
+};
+
+type State = {
+  clockName: string;
+  today: Date;
+};
+
+export class Clock extends Component<Props, State> {
+  state = {
+    clockName: this.props.name,
+    today: new Date(),
+  };
+
+  timerId = 0;
+
+  counterId = 0;
+
+  componentDidMount(): void {
+    this.timerId = window.setInterval(() => {
+      this.setState({ clockName: getRandomName() });
+    }, 3300);
+
+    this.counterId = window.setInterval(() => {
+      const today = new Date();
```

Comment:
why do u need this? U have already have "today" in state.

Reply:
Agree

Reply:
Agree